GitUI/CommandsDialogs/RevisionDiff.cs

```diff
@@ -293,6 +297,14 @@ private void DiffContextMenu_Opening(object sender, CancelEventArgs e)
             blameToolStripMenuItem.Enabled = isExactlyOneItemSelected;
             resetFileToToolStripMenuItem.Enabled = !isCombinedDiff;
 
+            this.diffEditFileToolStripMenuItem.Visible =
+               this.diffDeleteFileToolStripMenuItem.Visible =
+               isExactlyOneItemSelected && !DiffFiles.SelectedItem.IsSubmodule && selectedRevisions[0].IsArtificial();
+
+            this.diffToolStripSeparator13.Visible = isExactlyOneItemSelected &&
+                (!DiffFiles.SelectedItem.IsSubmodule && selectedRevisions[0].IsArtificial() ||
+                DiffFiles.SelectedItem.IsSubmodule && selectedRevisions[0].Guid == GitRevision.UnstagedGuid);
+
```

Comment:
Similar to my https://github.com/gitextensions/gitextensions/pull/4106#pullrequestreview-72685780 lets move the checks out to `RevisionDiffController` so the UI layer is dumb and the logic can be tested.